src/shared/vscode/defaultWindow.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Dialog, InputBox, Message, Progress, StatusBar, Window } from './window'
+
+export class DefaultWindow implements Window {
+    public readonly statusBar = new DefaultStatusBar()
+    public readonly inputBox = new DefaultInputBox()
+    public readonly message = new DefaultMessage()
+    public readonly progress = new DefaultProgress()
+    public readonly dialog = new DefaultDialog()
+}
+
+export class DefaultStatusBar implements StatusBar {
+    private readonly defaultTimeout = 2000
+
+    public setMessage(message: string, hideAfterTimeout: number = this.defaultTimeout): vscode.Disposable {
+        return vscode.window.setStatusBarMessage(message, hideAfterTimeout)
+    }
+}
+
+export class DefaultInputBox implements InputBox {
+    public show(options?: vscode.InputBoxOptions, token?: vscode.CancellationToken): Thenable<string | undefined> {
+        return vscode.window.showInputBox(options, token)
+    }
+}
+
+export class DefaultMessage implements Message {
+    public showError(message: string, ...items: string[]): Thenable<string | undefined> {
+        return vscode.window.showErrorMessage(message, ...items)
+    }
+}
+
+export class DefaultProgress implements Progress {
+    public show<R>(
+        options: vscode.ProgressOptions,
+        task: (
+            progress: vscode.Progress<{ message?: string; increment?: number }>,
+            token: vscode.CancellationToken
+        ) => Thenable<R>
+    ): Thenable<R> {
+        return vscode.window.withProgress(options, task)
+    }
+}
+
+export class DefaultDialog implements Dialog {
+    public showOpen(options: vscode.OpenDialogOptions): Thenable<vscode.Uri[] | undefined> {
+        return vscode.window.showOpenDialog(options)
+    }
+
+    public showSave(options: vscode.SaveDialogOptions): Thenable<vscode.Uri | undefined> {
```

Comment:
Looking at how this is implemented in https://github.com/aws/aws-toolkit-vscode/pull/1080 , I'm thinking I'd rather see this as mapping directly to the VS Code schema. Instead of `window.Dialog.showSave`, I'd rather see `window.showSaveDialog`. 

Goes the same for everything else here.

Reply:
Yeah I considered how much to stray from the original lumping of functions into namespaces. The difficulty with that approach is in discovering the inputs and outputs you can get/set. The namespaces are very large. Additionally, splitting them up into separate classes makes it so you can inject only the subset of window components you're using.

How would you suggest setting the user input for a save dialog?

Right now it's 

`window.dialog.saveSelection = 'blah.jpg'`

My thought was that this would be harder to discover among the many other properties attached to `window`.

`window.saveSelection = 'blah.jpg'`

Other options:
* `window.showSaveDialog(...)` is the function, but you use `window.dialogOptions.saveSelection` to get and set properties of it
* Attach the properties directly to the functions
  * `window.showSaveDialog.saveSelection`
* Support both call patterns

Reply:
Currently going with 

* Function call is `window.showSaveDialog(...)`
* Inputs are set up via `new FakeWindow(dialog: { saveSelection: 'foo' })`
* Outputs are retrieved via `window.dialog.{ saveOptions }`